### ISSUE TITLE
Features suggestion to add a third state to NVDA+7 (Focus moves navigator object)

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6746]
Using NVDA+7 to hold the review cursor on a piece of text while setting values in a dialog box, it occured to me that I generally leave it "on", but occasionally want to turn it "off".  Generally when I turn it off, what I really want, is to turn it off, while I'm on the current screen / in the current window, and then to set it back to "on".  I wondered then, whether it is worth changing NVDA+7 toggle to add a third state "Off within current window", which would turn it off, but as soon as focus moves to a new window, revert the state to "on" again.

**Comment by leonardder** [comment]
I tend to agree with this, but I wonder what to do with other hotkeys, 
such as nvda+6, nvda+5, etc. Should those also have such a third state? 
I can imagine that one could also find a use case to disable speech 
temporarily, for example.

**Comment by Qchristensen** [comment]
That's true, an argument could be made for both of those to also have an "off within the current window" state.

**Comment by feerrenrut** [comment]
Could it become annoying to have to toggle through three states?

**Comment by Qchristensen** [comment]
Potentially, although how often do most users need to change these settings?

Theoretically, if that was an issue, someone so inspired could create an addon similar to the "No beeps speech mode" one (which removes that option from the NVDA+S toggle).

**Comment by Brian1Gaff** [comment]
Not for me, but then I seem to recall in the past we had more anyway.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by Adriani90** [comment]
Is this not already possible with profiles?

**Comment by Qchristensen** [comment]
It presumably could be done with profiles, however that would leave it ALWAYS off in a particular program (or when a profile is active).  My use case was more around wanting to navigate to a control, toggle NVDA+7 so the review cursor stays there, go where I need to enter data and be able to read the text at the review cursor.  For instance, to see whether there is a key conflict when assigning shortcut keys in Word's Customise ribbon options, (probably particularly if assigning multiple shortcuts).

In that situation, having put the review cursor onto the line which indicates any conflicts, and set all the desired keystrokes, I press ENTER to activate the OK button.  Not only am I (logically) finished with that placement of the review cursor, but in this particular instance, the Window it was in no longer exists.  The review cursor won't move, because I told it not to.  The review cursor now reads "unknown' and I can't move it aside from toggling NVDA+7 and moving the focus.  What I really intended in that situation, was "don't move the review cursor while the focus is in this window".
